packages/migrate-config/src/migrate-config.js

```diff
@@ -0,0 +1,882 @@
+				});
+			} else {
+				migration.imports
+					.get("@eslint/compat")
+					.bindings.push("fixupConfigRules");
+			}
+
+			extendsCallExpression = b.callExpression(
+				b.identifier("fixupConfigRules"),
+				[extendsCallExpression],
+			);
+		}
+
+		// if there are either files or ignores, map so the resulting object has files and ignores
+		if (files || ignores) {
+			extendsCallExpression = b.callExpression(
+				b.memberExpression(extendsCallExpression, b.identifier("map")),
+				[
+					b.arrowFunctionExpression(
+						[b.identifier("config")],
+						b.objectExpression([
+							b.spreadElement(b.identifier("config")),
+							...(files
+								? [
+										b.property(
+											"init",
+											b.identifier("files"),
+											files,
+										),
+									]
+								: []),
+							...(ignores
+								? [
+										b.property(
+											"init",
+											b.identifier("ignores"),
+											ignores,
+										),
+									]
+								: []),
+						]),
+					),
+				],
+			);
+		}
+
+		configArrayElements.push(b.spreadElement(extendsCallExpression));
+	}
+
+	// Copy over `noInlineConfig` and `reportUnusedDisableDirectives`
+	const linterOptions = createLinterOptions(config);
+	if (linterOptions) {
+		properties.push(
+			b.property("init", b.identifier("linterOptions"), linterOptions),
+		);
+	}
+
+	// Create `languageOptions` from `env`, `globals`, `parser`, and `parserOptions`
+	const languageOptions = createLanguageOptions(migration, config);
+	if (languageOptions) {
+		properties.push(
+			b.property(
+				"init",
+				b.identifier("languageOptions"),
+				languageOptions,
+			),
+		);
+	}
+
+	// Copy over everything that stays the same - `settings`, `rules`, `processor`
+	keysToCopy.forEach(key => {
+		if (config[key]) {
+			const propertyValue =
+				typeof config[key] === "object"
+					? createAST(config[key])
+					: b.literal(config[key]);
+			properties.push(
+				b.property("init", b.identifier(key), propertyValue),
+			);
+		}
+	});
+
+	configArrayElements.push(b.objectExpression(properties));
+
+	return configArrayElements;
+}
+
+/**
+ * Migrates an eslintrc config to flat config format.
+ * @param {Config} config The eslintrc config to migrate.
+ * @param {Object} [options] Options for the migration.
+ * @param {"module"|"commonjs"} [options.sourceType] The module type to use.
+ * @returns {{code:string,messages:Array<string>,imports:Map<string,MigrationImport>}} The migrated config and
+ * any messages to display to the user.
+ */
+export function migrateConfig(config, { sourceType = "module" } = {}) {
+	const migration = new Migration(config);
+	const body = [];
+	const configArrayElements = [
+		...migrateConfigObject(
+			migration,
+			/** @type {ConfigOverride} */ (config),
+		),
+	];
+
+	// add any overrides
+	if (config.overrides) {
+		config.overrides.forEach(override => {
+			configArrayElements.push(
+				...migrateConfigObject(migration, override),
+			);
+		});
+	}
+
+	if (config.extends) {
```

Comment:
```suggestion
	if (config.extends || config.overrides?.some(override => override.extends)) {
```

Example:

```json
{
  "overrides": [{
    "files": "src/app/**",
    "extends": ["airbnb"]
  }]
}
```

Currently, migrate-config for this creates a config that's missing flat compat init:

```js
export default [{}, ...compat.extends("airbnb").map(config => ({
    ...config,
    files: ["src/app/**"],
})), {
    files: ["src/app/**"],
}];
```